packages/sitecore-jss/src/site/graphql-error-handling-service.ts

```diff
@@ -0,0 +1,104 @@
+import { GraphQLClient, GraphQLRequestClient } from '../graphql';
+import { siteNameError } from '../constants';
+import debug from '../debug';
+
+// The default query for request error handling
+const defaultQuery = /* GraphQL */ `
+  query ErrorHandlingQuery($siteName: String!, $language: String!) {
+    site {
+      siteInfo(site: $siteName) {
+        errorHandling(language: $language) {
+          notFoundPagePath
+          serverErrorPagePath
+        }
+      }
+    }
+  }
+`;
+
+export type GraphQLErrorHandlingServiceConfig = {
+  /**
+   * Your Graphql endpoint
+   */
+  endpoint: string;
+  /**
+   * The API key to use for authentication
+   */
+  apiKey: string;
+  /**
+   * The JSS application name
+   */
+  siteName: string;
+  /**
+   * The language
+   */
+  language: string;
+};
+
+export type ErrorHandlingType = { notFoundPagePath: string; serverErrorPagePath: string };
+
+/**
+ * The schema of data returned in response to error pages link request
+ */
+export type ErrorHandlingQueryResult = { site: { siteInfo: { errorHandling: ErrorHandlingType } } };
+
+/**
+ * Service that fetch the error handling data using Sitecore's GraphQL API.
+ */
+export class GraphQLErrorHandlingService {
```

Comment:
I'm struggling to decide about the name, but for me seems like `ErrorHandling` is a very abstract name, we should have in this case something like:
`GraphQLErrorHandlingPagesService` and -> `graphql-error-handling-pages-service`